test/distributed/org/apache/cassandra/distributed/test/repair/AutoRepairSchedulerTest.java

```diff
@@ -61,28 +61,35 @@ public static void init() throws IOException
         // Create SimpleDateFormat object with the given pattern
         sdf = new SimpleDateFormat(pattern);
         sdf.setLenient(false);
-        cluster = Cluster.build(3).withConfig(config -> config
-                                                        .set("auto_repair",
-                                                             ImmutableMap.of(
-                                                             "repair_type_overrides",
-                                                             ImmutableMap.of(AutoRepairConfig.RepairType.FULL.getConfigName(),
-                                                                             ImmutableMap.of(
-                                                                             "initial_scheduler_delay", "5s",
-                                                                             "enabled", "true",
-                                                                             "parallel_repair_count", "1",
-                                                                             "parallel_repair_percentage", "0",
-                                                                             "min_repair_interval", "1s"),
-                                                                             AutoRepairConfig.RepairType.INCREMENTAL.getConfigName(),
-                                                                             ImmutableMap.of(
-                                                                             "initial_scheduler_delay", "5s",
-                                                                             "enabled", "true",
-                                                                             "parallel_repair_count", "1",
-                                                                             "parallel_repair_percentage", "0",
-                                                                             "min_repair_interval", "1s"))))
-                                                        .set("auto_repair.enabled", "true")
-                                                        .set("auto_repair.global_settings.repair_by_keyspace", "true")
-                                                        .set("auto_repair.repair_task_min_duration", "0s")
-                                                        .set("auto_repair.repair_check_interval", "10s")).start();
+        // Configure a 3-node cluster with num_tokens: 4 and auto_repair enabled with parallel_repair_count: 2 to
```

Comment:
I made a few changes to the dtest to exercise the change of this behavior, but was also good to ensure that all replicas did in fact get their turn.  You should see the following logged occasionally while the test runs:

> INFO  [node3_AutoRepair-Repair-full:1] node3 2025-03-12 20:29:57,093 AutoRepairUtils.java:619 - Deferring repair because replica 6d194555-f6eb-41d0-c000-000000000003 (/127.0.0.3:7012) with shared ranges for 1 keyspace(s) (e.g. distributed_test_keyspace) is currently taking its turn

I'd like to write an explicit test for this but still racking my brain about how to go about doing this.  Will focus on that tomorrow.